### PR TITLE
uucore/buf_copy: bug fixes for compatibility with uutils core utilities

### PR DESCRIPTION
**Pull request by DaringCuteSeal**
This PR addresses several issues that previously hindered the usage of the new buffer-based copying module (#6964) in several core utilities (notably `cp`, `cat`, `yes`, and `install`).

### CODE REVIEW

File: src/uucore/src/lib/features/buf_copy.rs [hunk 1]
```diff
@@ -9,27 +9,55 @@
 //! used by utilities to work around the limitations of Rust's `fs::copy` which
 //! does not handle copying special files (e.g pipes, character/block devices).
 
-use crate::error::{UError, UResult};
-use nix::unistd;
+use crate::error::UError;
+use std::io::{Read, Write};
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
```
Comment (sylvestre) [inline src/uucore/src/lib/features/buf_copy.rs:15]: it is starting to look messy with all the cfg
maybe the .rs file per platform ?
for example:
src/uu/tail/src/platform/

**Review by sylvestre — CHANGES_REQUESTED** [review]
needs some refactoring. Too many cfg

**Comment by DaringCuteSeal** [comment]
oh right on that, wait

**Comment by DaringCuteSeal** [comment]
@sylvestre how's it

File: src/uucore/src/lib/features/buf_copy.rs [hunk 2]
```diff
@@ -9,283 +9,53 @@
...
-    use super::*;
+    #[cfg(unix)]
     use crate::pipes;
+    #[cfg(unix)]
+    use std::fs::OpenOptions;
+    #[cfg(unix)]
+    use std::{
+        io::{Seek, SeekFrom},
+        thread,
+    };
```
Comment (sylvestre) [inline src/uucore/src/lib/features/buf_copy.rs:38]:
```suggestion
    #[cfg(unix)]
    {
    use crate::pipes;
    use std::fs::OpenOptions;
    use std::{
        io::{Seek, SeekFrom},
        thread,
    };
    }
```

Comment (sylvestre) [reply src/uucore/src/lib/features/buf_copy.rs:38]: I think it works :)

File: src/uucore/src/lib/features/buf_copy.rs [hunk 3]
```diff
@@ -9,283 +9,53 @@
...
+
+    #[cfg(any(target_os = "linux", target_os = "android"))]
```
Comment (sylvestre) [inline src/uucore/src/lib/features/buf_copy.rs:40]: same change here

File: src/uucore/src/lib/features/buf_copy/common.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+use crate::error::UError;
```
Comment (sylvestre) [inline src/uucore/src/lib/features/buf_copy/common.rs:1]: please add the license header here

**Comment by sylvestre** [comment]
much better, thanks :)
just a few comments now

**Comment by DaringCuteSeal** [comment]
there, should be ok now

**Comment by sylvestre** [comment]
thanks!

**Pull request merged**